diff --git a/Makefile.gnu b/Makefile.gnu
index a8c6c31..968d52e 100644
--- a/Makefile.gnu
+++ b/Makefile.gnu
@@ -16,8 +16,8 @@ LIBRARIES = -lstdc++
 MODULES = $(SRCS:.c=.o)
 MODULES := $(MODULES:.cpp=.o)
 CFLAGS ?= -O3 -fPIC -fexceptions -fvisibility=hidden
-override CFLAGS += $(INCLUDE) -D__ANSI__ -I/usr/include/jxrlib $(shell pkg-config --cflags OpenEXR libopenjp2 libraw libpng libtiff-4 libwebp libwebpmux zlib)
-override LDFLAGS += -ljpeg -ljpegxr -ljxrglue $(shell pkg-config --libs OpenEXR libopenjp2 libraw libpng libtiff-4 libwebp libwebpmux zlib)
+override CFLAGS += $(INCLUDE) -D__ANSI__ -I/usr/include/jxrlib $(shell pkg-config --cflags libpng libtiff-4 libwebp libwebpmux zlib)
+override LDFLAGS += -ljpeg $(shell pkg-config --libs libpng libtiff-4 libwebp libwebpmux zlib)
 
 ifeq ($(shell sh -c 'uname -m 2>/dev/null || echo not'),x86_64)
 	override CFLAGS += -fPIC
diff --git a/Source/FreeImage/Plugin.cpp b/Source/FreeImage/Plugin.cpp
index 636b6c7..2412be3 100644
--- a/Source/FreeImage/Plugin.cpp
+++ b/Source/FreeImage/Plugin.cpp
@@ -246,7 +246,7 @@ FreeImage_Initialise(BOOL load_local_plugins_only) {
 			s_plugins->AddNode(InitPNM, NULL, "PBM", "Portable Bitmap (ASCII)", "pbm", "^P1");
 			s_plugins->AddNode(InitPNM, NULL, "PBMRAW", "Portable Bitmap (RAW)", "pbm", "^P4");
 			s_plugins->AddNode(InitPCD);
-			s_plugins->AddNode(InitPCX);
+//			s_plugins->AddNode(InitPCX);
 			s_plugins->AddNode(InitPNM, NULL, "PGM", "Portable Greymap (ASCII)", "pgm", "^P2");
 			s_plugins->AddNode(InitPNM, NULL, "PGMRAW", "Portable Greymap (RAW)", "pgm", "^P5");
 			s_plugins->AddNode(InitPNG);
@@ -270,15 +270,15 @@ FreeImage_Initialise(BOOL load_local_plugins_only) {
  			s_plugins->AddNode(InitG3);
 #endif
 			s_plugins->AddNode(InitSGI);
-			s_plugins->AddNode(InitEXR);
-			s_plugins->AddNode(InitJ2K);
-			s_plugins->AddNode(InitJP2);
+//			s_plugins->AddNode(InitEXR);
+//			s_plugins->AddNode(InitJ2K);
+//			s_plugins->AddNode(InitJP2);
 			s_plugins->AddNode(InitPFM);
 			s_plugins->AddNode(InitPICT);
-			s_plugins->AddNode(InitRAW);
+//			s_plugins->AddNode(InitRAW);
 			s_plugins->AddNode(InitWEBP);
 #if !(defined(_MSC_VER) && (_MSC_VER <= 1310))
-			s_plugins->AddNode(InitJXR);
+//			s_plugins->AddNode(InitJXR);
 #endif // unsupported by MS Visual Studio 2003 !!!
 			
 			// external plugin initialization
diff --git a/Source/FreeImage/PluginTIFF.cpp b/Source/FreeImage/PluginTIFF.cpp
index e1e62e0..150784a 100644
--- a/Source/FreeImage/PluginTIFF.cpp
+++ b/Source/FreeImage/PluginTIFF.cpp
@@ -39,7 +39,7 @@
 #include "Utilities.h"
 #include <tiffio.h>
 #include "../Metadata/FreeImageTag.h"
-#include <OpenEXR/half.h>
+#include "../OpenEXR/Half/half.h"
 
 #include "FreeImageIO.h"
 #include "PSDParser.h"
diff --git a/genfipsrclist.sh b/genfipsrclist.sh
index 4c0cc87..b14a4c0 100644
--- a/genfipsrclist.sh
+++ b/genfipsrclist.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-DIRLIST="Wrapper/FreeImagePlus"
+DIRLIST="Source/OpenEXR Source/OpenEXR/Half Source/OpenEXR/Iex Source/OpenEXR/IlmImf Source/OpenEXR/IlmThread Source/OpenEXR/Imath Source/OpenEXR/IexMath Wrapper/FreeImagePlus"
 
 
 echo "VER_MAJOR = 3" > fipMakefile.srcs
diff --git a/gensrclist.sh b/gensrclist.sh
index db59efa..e6b4634 100644
--- a/gensrclist.sh
+++ b/gensrclist.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-DIRLIST=". Source Source/Metadata Source/FreeImageToolkit"
+DIRLIST=". Source Source/Metadata Source/FreeImageToolkit Source/OpenEXR Source/OpenEXR/Half Source/OpenEXR/Iex Source/OpenEXR/IlmImf Source/OpenEXR/IlmThread Source/OpenEXR/Imath Source/OpenEXR/IexMath"
 
 echo "VER_MAJOR = 3" > Makefile.srcs
 echo "VER_MINOR = 17.0" >> Makefile.srcs
